tasks/webcrawler.js

```diff
@@ -242,16 +242,29 @@ Crawler.prototype.isUrlRelative = function(uri) {
  * @returns {Boolean} TRUE if url is the same origin as base.
  */
 Crawler.prototype.isUrlSameOrigin = function (uri) {
-        var baseUri = URI(this.baseUrl);
+    var baseUri = URI(this.baseUrl);
 
-        return baseUri.normalizeProtocol().protocol() === uri.normalizeProtocol().protocol() &&
-                     baseUri.normalizeHostname().host() === uri.normalizeHostname().host();
+    return baseUri.normalizeProtocol().protocol() === uri.normalizeProtocol().protocol() &&
+        baseUri.normalizeHostname().host() === uri.normalizeHostname().host();
+};
+
+/**
+ * Render a snapshot of the page with phantom.
+ *
+ * @param {Object} page The page object.
+ * @param {String} path The relative path of the file to snapshot.
+ */
+Crawler.prototype.createSnapshot = function (page, path) {
+    // Only render if options set.
+    if (this.options.render && undefined !== page) {
+        page.render(this.options.renderDir + "/" + this.getStaticFilename(path) + ".png");
+    }
 };
 
 /**
  * Crawl a url.
  *
- * @param {String} url - The absolute or relative url to crawl.
```

Comment:
Son of a... Not enough coffee this morning. Or maybe I need  more sugar? Either way I'm glad that I review my own code.